File: app/src/main/java/com/weathercock/android_cleanarchitecture/di/module/DataModule.java
package com.weathercock.android_cleanarchitecture.di.module;

import android.content.Context;

import com.weathercock.android_cleanarchitecture.data.repository.AboutRepositoryImpl;
import com.weathercock.android_cleanarchitecture.data.repository.CityRepositoryImpl;
import com.weathercock.android_cleanarchitecture.data.repository.WeatherRepositoryImpl;
import com.weathercock.android_cleanarchitecture.domain.repository.AboutRepository;
import com.weathercock.android_cleanarchitecture.domain.repository.CityRepository;
import com.weathercock.android_cleanarchitecture.domain.repository.WeatherRepository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class DataModule {

    @Provides
    @Singleton
    public CityRepository provideCityRepository() {
        return new CityRepositoryImpl();
    }

    @Provides
    @Singleton
    public WeatherRepository provideWeatherRepository() {
        return new WeatherRepositoryImpl();
    }

    @Provides
    @Singleton
    public AboutRepository provideAboutRepository(Context context) {
        return new AboutRepositoryImpl(context);
    }

}

File: data/repository/src/main/java/com/kazakago/blueprint/data/repository/di/DataModule.kt
package com.kazakago.blueprint.data.repository.di

import com.kazakago.blueprint.data.repository.hierarchy.about.AboutRepositoryImpl
import com.kazakago.blueprint.data.repository.hierarchy.city.CityRepositoryImpl
import com.kazakago.blueprint.data.repository.hierarchy.weather.WeatherRepositoryImpl
import com.kazakago.blueprint.domain.repository.hierarchy.about.AboutRepository
import com.kazakago.blueprint.domain.repository.hierarchy.city.CityRepository
import com.kazakago.blueprint.domain.repository.hierarchy.weather.WeatherRepository
import org.koin.dsl.module

val dataModule = module {
    single<CityRepository> { CityRepositoryImpl(get()) }
    single<AboutRepository> { AboutRepositoryImpl(get()) }
    single<WeatherRepository> { WeatherRepositoryImpl(get()) }
}
